required/latex-lab/latex-lab-float.dtx

```diff
@@ -203,73 +252,147 @@
+  }  
+\cs_generate_variant:Nn \@@_float_container_new:nN {eN}  
+%    \end{macrocode}
+% \end{macro}
 % \begin{macro}{\@@_float_init_collect:}
 % This initializes a container structure for every float type.
 % It can be used more than once in a document, this allows to have e.g.
 % chapter wise containers.
 %    \begin{macrocode}
 \cs_new_protected:Npn\@@_float_init_collect:
-  { 
+  {
     \bool_if:NT\g_@@_float_sect_bool
       {
         \seq_map_inline:Nn\g_@@_float_types_seq
           {
-            \tag_struct_begin:n{tag=##1s,stash}
-            \prop_gput:Nne\g_@@_float_sect_prop {##1-struct}{\int_use:N\c@g__tag_struct_abs_int} 
-            \tag_struct_end:
-          }  
-      }   
+            \@@_float_container_new:nN {##1}\l_@@_tmpa_tl
+          }
+      }
   }
 %    \end{macrocode}
 % \end{macro}
-% 
-% 
+%
+%
 % \begin{macro}{\@@_float_stop_sect:}
 % This pushes out the floats. For every type is checks if there is actually
 % a float of this type and then writes out the container structure.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_float_stop_sect:
   {
-    \bool_if:NT\g_@@_float_sect_bool
+    \seq_map_inline:Nn\g_@@_float_types_seq
+      {
+        \prop_get:NnNT\g_@@_float_sect_prop{##1-used}\l_@@_float_tmpa_tl
+          {
+            \prop_get:NnNF\g_@@_float_sect_prop{##1-struct}\l_@@_float_tmpb_tl
+             {
+               \@@_float_container_new:eN {##1}\l_@@_float_tmpb_tl
+             }
+            \exp_args:Ne
+            \tag_struct_use_num:n{\l_@@_float_tmpb_tl}
+            \prop_gremove:Nn \g_@@_float_sect_prop{##1-used}                
+          }
+      }
+  }
+%    \end{macrocode}
+% \end{macro}
+% \begin{macro}{flush/float}
+% This is a key for \cs{tagpdfsetup} to flush out the collected floats.
+% The value allows to set to which level the create Sect belongs.
+% So \texttt{section} will close all previous Sect until the section level
+% and create a new section structure.
+%    \begin{macrocode}
+\keys_define:nn { __tag / setup }
+ {
+   float/flush .code:n =
      {
-      \seq_map_inline:Nn\g_@@_float_types_seq
-        {
-          \prop_get:NnNT\g_@@_float_sect_prop{##1-used}\l_@@_tmpa_tl
-            {
-              \exp_args:Ne
-              \tag_struct_use_num:n{\prop_item:Nn\g_@@_float_sect_prop{##1-struct}}
-              \prop_gremove:Nn \g_@@_float_sect_prop{##1-used}
-            }  
-        }  
-     }   
-  } 
+       \par
+       \cs_if_exist:cTF{toclevel@#1}
+        { \UseTaggingSocket{sec/end}{\int_eval:n{\use:c{toclevel@#1}}}}
+        { \UseTaggingSocket{sec/end}{\int_eval:n{#1}}}
+       \@@_float_stop_sect:
+       \@@_float_init_collect:
+     },
+   float/flush .default:n = \@currentseclevel
+ }
 %    \end{macrocode}
 % \end{macro}
 
-% \begin{macro}{flush-floats}
-% This is a key for \cs{tagtool} to flush out the collected floats. 
-% The value allows to set to which level the create Sect contains. 
+% \begin{macro}{flush-floats (deprecated)}
```

Comment:
Is there a reason why the code is repeated in that case? Wouldn't it be cleaner or at least more readable if it is turned into a meta key or something?

Reply:
well actually I would like to remove it in the next clean up round so I didn't bother.  But I can change to a key.